src/lib/dialog/dialog-config.ts

```diff
@@ -14,6 +14,9 @@ export class MdDialogConfig {
   /** The ARIA role of the dialog element. */
   role: DialogRole = 'dialog';
 
+  /** Whether the escape key should close the dialog. */
+  escapeToClose = true;
```

Comment:
Let's just make this `disableClose` and have it apply to both escape and backdrop click, since you're unlikely to want to allow one but not the other.